packages/core/src/modules/message-pìckup/protocol/v2/V2MessagePickupProtocol.ts

```diff
@@ -4,26 +4,36 @@ import type { AgentMessageReceivedEvent } from '../../../../agent/Events'
 import type { FeatureRegistry } from '../../../../agent/FeatureRegistry'
 import type { InboundMessageContext } from '../../../../agent/models/InboundMessageContext'
 import type { DependencyManager } from '../../../../plugins'
-import type { MessageRepository } from '../../../../storage/MessageRepository'
 import type { EncryptedMessage } from '../../../../types'
-import type { PickupMessagesProtocolOptions, PickupMessagesProtocolReturnType } from '../MessagePickupProtocolOptions'
+import type { MessagePickupRepository } from '../../storage/MessagePickupRepository'
+import type {
+  DeliverMessagesProtocolOptions,
+  DeliverMessagesProtocolReturnType,
+  PickupMessagesProtocolOptions,
+  PickupMessagesProtocolReturnType,
+  SetLiveDeliveryModeProtocolOptions,
+  SetLiveDeliveryModeProtocolReturnType,
+} from '../MessagePickupProtocolOptions'
 
 import { EventEmitter } from '../../../../agent/EventEmitter'
 import { AgentEventTypes } from '../../../../agent/Events'
 import { MessageSender } from '../../../../agent/MessageSender'
 import { OutboundMessageContext, Protocol } from '../../../../agent/models'
 import { InjectionSymbols } from '../../../../constants'
 import { Attachment } from '../../../../decorators/attachment/Attachment'
-import { AriesFrameworkError } from '../../../../error'
 import { injectable } from '../../../../plugins'
 import { ConnectionService } from '../../../connections'
+import { verkeyToDidKey } from '../../../dids/helpers'
```

Comment:
Not sure if this util already existed, but we should probably rename it to `publicKeyBase58ToDidKey`